networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/peer/DisconnectReason.java

```diff
@@ -36,9 +37,11 @@
   }
 
   public static Optional<DisconnectReason> fromReasonCode(final UInt64 reasonCode) {
+    // Pull the min matching value so that we deterministically map code to reason even if some
+    // reasons share the same code
     return Stream.of(values())
         .filter(reason -> reason.getReasonCode().equals(reasonCode))
-        .findAny();
+        .min(Comparator.comparing(Enum::name));
```

Comment:
`values()` always iterates the enum in declared order and it's a sequential stream so `findAny` would always find the first of them.  So unless I'm missing something I think we're just doing extra work here (always iterating every value instead of short circuiting) for no benefit.

Reply:
Yeah - when testing, it seemed to be deterministic either way.  But I wasn't sure if the behavior was guaranteed or not.  

According to [this article](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html) they are deterministically ordered.  So, I can just revert this.